fhir-persistence/src/test/java/com/ibm/fhir/persistence/test/common/AbstractCanonicalTest.java

```diff
@@ -0,0 +1,767 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.persistence.test.common;
+
+import static com.ibm.fhir.model.type.String.string;
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertNotNull;
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.ibm.fhir.model.resource.CarePlan;
+import com.ibm.fhir.model.resource.Library;
+import com.ibm.fhir.model.resource.Measure;
+import com.ibm.fhir.model.resource.Resource;
+import com.ibm.fhir.model.test.TestUtil;
+import com.ibm.fhir.model.type.Canonical;
+import com.ibm.fhir.model.type.Reference;
+import com.ibm.fhir.model.type.Uri;
+
+/**
+ *  This class tests the persistence layer support for the FHIR canonical search parameters.
+ */
+public abstract class AbstractCanonicalTest extends AbstractPersistenceTest {
+    private static CarePlan savedCarePlan1;
+    private static CarePlan savedCarePlan2;
+    private static CarePlan savedCarePlan3;
+    private static Measure savedMeasure1;
+    private static Measure savedMeasure2;
+    private static Measure savedMeasure3;
+    private static Measure savedMeasure4;
+    private static Library savedLibrary1;
+    private static Library savedLibrary2;
+    private static Library savedLibrary3;
+    private static String uuid = UUID.randomUUID().toString();
+
+    /**
+     * Loads up and saves a bunch of resources with various canonical references to one another
+     */
+    @BeforeClass
+    public void createResources() throws Exception {
+        CarePlan carePlan = TestUtil.readExampleResource("json/ibm/minimal/CarePlan-1.json");
+        Measure measure = TestUtil.readExampleResource("json/ibm/minimal/Measure-1.json");
+        Library library = TestUtil.readExampleResource("json/ibm/minimal/Library-1.json");
```

Comment:
FYI I created TestUtil.getMinimalResource(ResourceType) a while back as a convenience method.  It does the same thing anyway, so doesn't really matter; just wanted to mention it.

If we did use getMinimalResource consistently we could potentially change the implementation to keep the minimal resources around in memory so that we don't need to parse the files for each use.

Reply:
changed to use getMinimalResource

Reply:
cool, I decided to just whip up these changes since they were on my mind:  https://github.com/IBM/FHIR/pull/2701
I don't think it will have much impact, but it does mean we can avoid re-reading these same minimal resources over and over and could eventually add up to some savings.